core/domain/state_domain.py

```diff
@@ -1551,6 +1599,20 @@ def update_content_ids_to_audio_translations(
                 content_ids_to_audio_translations_dict.iteritems())
         }
 
+    def update_written_translations(self, written_translations_dict):
```

Comment:
Can we pass the object instead?

Reply:
This will break consistency as we use to pass the dict in every place and for every part including exploration, skill, topics etc. Also, I think from the service layer we are just aware of the State object and we are passing the dict such that it should create a domain object out of it from one of the classes present within the state_domain.py

Also, this will change the format of writing a test for updating WrittenTranslations keeping all other parts as is. Let me know if you want me to pas the object instead and change the test files accordingly OR create an issue for the same to have a PR changing this everywhere?

Reply:
I forgot to discuss this when we met, oops. 

Let's do both: pass an object, and file an issue to do the same for other places. In general we shouldn't pass dicts around.

(Has the dict passing been around for a long time or just recently? I don't recall this being a pattern that we used last time...)

Reply:
@seanlip, I've created an issue related to this at #6271.